### ISSUE TITLE
SSLClientCoordinator: SSL config is reloaded multiple times

### ISSUE THREAD

**Issue by cmcfarlen** [apache/trafficserver#11724]
If multiple ssl related config files are changed and then `traffic_ctl config reload` is executed, the ssl config will reload multiple times.

Reproduction:

1. Run traffic_server with a sni.yaml and ssl_multicert.config file configured
2. touch sni.yaml
3. `traffic_ctl config reload`
4. Observerve the log output:
```
[Aug 20 13:59:35.145] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/sni.yaml loading ...
[Aug 20 13:59:35.147] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/sni.yaml finished loading
[Aug 20 13:59:35.147] [ET_TASK 0] NOTE: ssl_multicert.config loading ...
[Aug 20 13:59:35.158] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/ssl_multicert.config finished loading
```
5. touch sni.yaml and ssl_multicert.config files
6. `traffic_ctl config reload`
7. Observer the log output:
```
[Aug 20 13:59:53.154] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/sni.yaml loading ...
[Aug 20 13:59:53.156] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/sni.yaml finished loading
[Aug 20 13:59:53.156] [ET_TASK 0] NOTE: ssl_multicert.config loading ...
[Aug 20 13:59:53.166] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/ssl_multicert.config finished loading
[Aug 20 13:59:53.166] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/sni.yaml loading ...
[Aug 20 13:59:53.167] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/sni.yaml finished loading
[Aug 20 13:59:53.168] [ET_TASK 0] NOTE: ssl_multicert.config loading ...
[Aug 20 13:59:53.174] [ET_TASK 0] NOTE: /opt/ats/etc/trafficserver/ssl_multicert.config finished loading
```

This happens because the callbacks setup in SSLClientCoordinator are independently attached and so the reconfigure function will be called for every updated config item specified.

This happens on the master branch, but also for 9.2.x

**Comment by JosiahWI** [comment]
Is this closed by #11723 @cmcfarlen @maskit?

**Comment by maskit** [comment]
No, we found the QUIC reload issue while we work on this one. With #11723 fixed, we'd see 4 "ssl_multicert.config loading ..." lines in total if QUIC is available.

**Comment by bryancall** [comment]
This was fixed by [PR #12892](https://github.com/apache/trafficserver/pull/12892) (the configuration reload token model). The root cause was exactly as described here: `SSLClientCoordinator::startup()` used a `ConfigUpdateHandler` and called `attach()` once per record, so the reconfigure callback fired independently for each changed SSL config file, producing duplicate reloads. That PR replaced the per-record attach mechanism with a single coordinated registration (`ConfigRegistry::register_record_config` under the key `ssl_client_coordinator`), so the reconfigure handler now runs once per reload regardless of how many SSL files changed. The follow-up [PR #13075](https://github.com/apache/trafficserver/pull/13075) added a `Test.AddConfigReload()` autest extension covering reload behavior. For reference, [PR #11723](https://github.com/apache/trafficserver/pull/11723) was a separate QUIC reload bug found while investigating this one and is not the fix here. The fix is on the master branch and is scheduled for the 11.0.0 release. Closing as resolved.
